lib/abr/ewma.js

```diff
@@ -54,7 +54,10 @@ shaka.abr.Ewma = function(halfLife) {
  */
 shaka.abr.Ewma.prototype.sample = function(weight, value) {
   var adjAlpha = Math.pow(this.alpha_, weight);
-  this.estimate_ = value * (1 - adjAlpha) + adjAlpha * this.estimate_;
+  var newEstimate = value * (1 - adjAlpha) + adjAlpha * this.estimate_;
+  // In rare circumstances newEstimate may come out to NaN. In these
```

Comment:
In these cases, it would be preferable to drop the sample (return early).  Adding to totalWeight_ in these cases will skew future samples.